pkg/networkmanager/interfaces.js

```diff
@@ -165,12 +165,34 @@
     const client = cockpit.dbus("org.freedesktop.NetworkManager", { superuser: "try" });
     self.client = client;
 
+    // set to false by default as newer API is backwards compatible
+    // and supports both "dns" and "dns-data" properties
+    // dns-data property was added in version 1.42.0
+    self.supports_dns_data = false;
+
+    function check_version_dns_data_support(version) {
+        if (version.length < 2) {
+            return false;
+        }
+
+        if (version[0] > 1) {
+            return true;
+        } else if (version[0] === 1 && version[1] >= 42) {
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22330-1fe4fb60-20250903-092011-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)